ghga_connector/core/api_calls.py

```diff
@@ -84,12 +85,14 @@ def initiate_multipart_upload(api_url: str, file_id: str) -> Tuple[str, int]:
 
     status_code = response.status_code
     if status_code != 200:
-        if status_code == 400:
-            raise NoUploadPossibleError(file_id=file_id)
-        if status_code == 403:
-            raise UserHasNoFileAccess(file_id=file_id)
-        if status_code == 404:
-            raise FileNotRegisteredError(file_id=file_id)
+        spec = {
+            400: {
+                "existingActiveUpload": lambda: NoUploadPossibleError(file_id=file_id),
+                "fileNotRegistered": lambda: FileNotRegisteredError(file_id=file_id),
+            },
+            403: {"noFileAccess": lambda: UserHasNoFileAccess(file_id=file_id)},
+        }
+        ResponseExceptionTranslator(spec=spec).handle(response=response)
```

Comment:
Looks good now. Maybe we should also include such a wrapper in the httpyexpect library, that hides the two step initialization process (initialising the mapping and then initializing the translator).